src/Codeception/Module/Db.php

```diff
@@ -205,156 +194,271 @@ class Db extends CodeceptionModule implements DbInterface
+    public $currentDatabase = self::DEFAULT_DATABASE;
 
-    /**
-     * @var array
-     */
-    protected $requiredFields = ['dsn', 'user', 'password'];
+    public function getDatabases()
+    {
+        $databases = [$this->currentDatabase => $this->config];
+
+        if (!empty($this->config['databases'])) {
+            foreach ($this->config['databases'] as $databaseKey => $databaseConfig) {
+                $databases[$databaseKey] = array_merge([
+                    'populate' => false,
+                    'cleanup' => false,
+                    'reconnect' => false,
+                    'dump' => null,
+                    'populator' => null,
+                ], $databaseConfig);
+            }
+        }
+        return $databases;
+    }
+    public function connectToDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->connect($databaseKey, $databaseConfig);
+        }
+    }
+    public function cleanUpDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->_cleanup($databaseKey, $databaseConfig);
+        }
+    }
+    public function populateDatabases($configKey)
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            if ($databaseConfig[$configKey]) {
+                $this->_loadDump($databaseKey, $databaseConfig);
+            }
+        }
+    }
+    public function readSqlForDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->readSql($databaseKey, $databaseConfig);
+        }
+    }
+    public function removeInsertedForDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->amConnectedToDatabase($databaseKey);
+            $this->removeInserted($databaseKey, $databaseConfig);
+        }
+    }
+    public function disconnectDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->disconnect($databaseKey, $databaseConfig);
+        }
+    }
+    public function reconnectDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            if ($databaseConfig['reconnect']) {
+                $this->disconnect($databaseKey, $databaseConfig);
+                $this->connect($databaseKey, $databaseConfig);
+            }
+        }
+    }
+
+    public function __get($name)
+    {
+        if ($name == 'driver') {
+            return $this->getDriver();
+        }
+        if ($name == 'dbh') {
+            return $this->getDbh();
+        }
+    }
+
+    public function getDriver()
+    {
+        return $this->drivers[$this->currentDatabase];
+    }
+    public function getDbh()
+    {
+        return $this->dbhs[$this->currentDatabase];
+    }
+
+    public function amConnectedToDatabase($databaseKey, $callback = null)
```

Comment:
Ok, it's not that clear what does the callback do.
Please add a docblock here with documentation. Inside documentation please add code examples of usage

Reply:
Fixed